src/mongo/db/storage/mmap_v1/mmap_windows.cpp

```diff
@@ -184,8 +184,8 @@ void* MemoryMappedFile::map(const char* filenameIn, unsigned long long& length)
     strncpy(filename, filenameIn, 255);
     filename[255] = 0;
     {
-        size_t len = strlen(filename);
-        for (size_t i = len - 1; i >= 0; i--) {
+        int len = strlen(filename);
```

Comment:
strlen really does return a size_t:

http://pubs.opengroup.org/onlinepubs/009695399/functions/strlen.html

In the event that len is zero, what should happen here? I don't think just avoiding this cleaning loop is sufficient. Does an empty file name actually make sense here?